dace/symbolic.py

```diff
@@ -5,12 +5,14 @@
 import pickle
 import re
 from typing import Any, Callable, Dict, Iterable, Optional, Set, Tuple, Union
-import warnings
 import numpy
 
 import sympy.abc
 import sympy.printing.str
 
+# FIXME(ben): do we need to add `packaging` as a dependency to DaCe?
```

Comment:
The alternative, `pkg_resources`, is deprecated. I do not believe that the standalone `packaging` module is included by default in a barebones Python installation, although other modules may have it in their dependencies (it is in my conda environment for some reason). I would add it to the dependencies to finish this PR.

Reply:
Pytest requires it, but I would still add it to the base DaCe dependencies:
```
Collecting packaging (from pytest>=7.0.0->pytest-xdist)
  Downloading packaging-24.1-py3-none-any.whl.metadata (3.2 kB)
  ```

Reply:
Yes, I was thinking that most likely one of our dependencies needs it anyway, but definitely better to explicitly add it. It's a rather small dependency.